vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -126,6 +127,17 @@ void AdsImpl::InitializeStep4(const Result result) {
 
   NotificationAllowedCheck(false);
 
+  //ads notifications don't sustain reboot, so remove all
+  auto ads_shown_history = client_->GetAdsShownHistory();
```

Comment:
This logic should only be called for Android, please see `AdsImpl::IsMobile`

Reply:
Implemented `RemoveAllNotificationsAfterUpdate`.